app_feup/lib/redux/action_creators.dart

```diff
@@ -223,6 +223,11 @@ ThunkAction<AppState> getUserExams(Completer<Null> action) {
         final AppExamsDatabase db = AppExamsDatabase();
         db.saveNewExams(exams);
       }
+
+      exams.sort((exam1, exam2) {
+        return exam1.date.compareTo(exam2.date);
+      });
+
```

Comment:
Can we move this to before saving to the database? This way we guarantee with one sort call that the exams will be sorted.

Reply:
Are SQLite databases guaranteed to preserve insertion order?

Reply:
From my testing, it appears that insertion order is preserved in SQLite. Based on that, I have applied your suggestions.